src/pages/SignUp.tsx

```diff
@@ -1,21 +1,159 @@
+
+  return (
+    <PageWrapper
+      left={
+        <div className="formWrapper">
+          <h2>SIGN UP</h2>
+          <form onSubmit={handleSubmit(onSubmitForm)}>
+            <InputField
+              label="name"
+              id="username"
+              autoComplete="off"
+              register={register('username', {
+                required: 'Your name is required'
+              })}
+              error={extractApiErrors({
+                apiErrors: error,
+                formErrors: errors,
+                errorToExtract: 'username'
+              })}
+              type="text"
+            />
+            <InputField
+              label="email"
+              id="email"
+              register={register('email', {
+                required: 'Your email is required',
+                pattern: {
+                  value: /\S+@\S+\.\S+/,
+                  message: 'Entered value does not match email format'
+                }
+              })}
+              error={extractApiErrors({
+                apiErrors: error,
+                formErrors: errors,
+                errorToExtract: 'email'
+              })}
+              type="email"
+            />
+            <InputField
+              label="password"
+              id="password"
+              placeholder="MIN. 8 CHARACTERS LONG"
+              register={register('password', {
+                required: 'Your password is required',
+                minLength: { value: 8, message: 'Min. 8 characters long' }
+              })}
+              error={extractApiErrors({
+                apiErrors: error,
+                formErrors: errors,
+                errorToExtract: 'password'
+              })}
+              type="password"
+            />
+            <InputField
+              label="confirm password"
+              id="password_confirmation"
+              placeholder="MIN. 8 CHARACTERS LONG"
+              register={register('password_confirmation', {
+                required: 'Your password confirmation is required',
+                minLength: { value: 8, message: 'Min. 8 characters long' },
+                validate: (value) =>
+                  value === passwordCurrentValue || 'The passwords do not match'
+              })}
+              error={extractApiErrors({
+                apiErrors: error,
+                formErrors: errors,
+                errorToExtract: 'password_confirmation'
+              })}
+              type="password"
+            />
+            <SelectField
+              label="gender"
+              id="gender"
+              register={register('gender', {
+                required: 'Your gender is required'
+              })}
+              error={errors.gender ? errors.gender.message : ''}
+              defaultValue=""
+            >
+              <option value="" disabled>
+                SELECT YOUR GENDER
+              </option>
+              {Object.values(genders).map((gender) => (
```

Comment:
Maybe You can destruct gender here, but is not really important

Reply:
✅